app/models/notification.rb

```diff
@@ -15,7 +15,7 @@ def self.for(recipient, opts={})
   end
 
   def self.notify(recipient, target, actor)
-    return nil unless target.respond_to? :notification_type
+    return nil unless target.respond_to? :notification_type || recipient != actor
```

Comment:
What the f? I didn't get that mention. The power of zoning randomly. What do you need exactly, @cmrd-senya ?